lib/puma/runner.rb

```diff
@@ -159,5 +166,13 @@ def start_server
       server.inherit_binder @launcher.binder
       server
     end
+
+    private
+
+    def ensure_output_directory_exists(path)
```

Comment:
Maybe something like:
```ruby
def ensure_output_directory_exists(path, io_name)
```
and pass `'STDOUT'` or `'STDERR'` to it, along with changing:
```ruby
raise "Cannot redirect #{io_name} to #{path}"